test/integration/helpers/helpers.go

```diff
@@ -58,14 +60,31 @@ func NewTestContext(t *testing.T, driver *neo4j.DriverWithContext) *TestContext
 	if err != nil {
 		t.Fatalf("failed to create Neo4j service: %v", err)
 	}
-	deps := &tools.ToolDependencies{DBService: databaseService}
+	analyticsService := getAnalyticsMock(t)
+	deps := &tools.ToolDependencies{DBService: databaseService, AnalyticsService: analyticsService}
 
 	tc.Service = databaseService
 	tc.Deps = deps
 
 	return tc
 }
 
+// getAnalyticsMock is used to mock the analytics service, for integration test purpose.
+func getAnalyticsMock(t *testing.T) *analytics_mocks.MockService {
```

Comment:
I think we already discussed whether disabling this could be an option instead. 

But I'm also thinking, since these are integration tests, what's the downside of having actual analytics being fired from the tests? Do you think we can have that in an isolated namespace/API key, or would it be too expensive?